core/src/main/scala/caliban/schema/Schema.scala

```diff
@@ -150,6 +148,70 @@ trait GenericSchema[R] extends DerivationSchema[R] with TemporalSchema {
         ObjectStep(name, fields(false, false).map { case (f, plan) => f.name -> plan(value) }.toMap)
     }
 
+  def field[V](
+    name: String,
+    description: Option[String] = None,
+    directives: List[Directive] = List.empty
+  ): PartiallyAppliedField[V] =
+    PartiallyAppliedField[V](name, description, directives)
+
+  def fieldWithArgs[V, A](
+    name: String,
+    description: Option[String] = None,
+    directives: List[Directive] = Nil
+  ): PartiallyAppliedFieldWithArgs[V, A] =
+    PartiallyAppliedFieldWithArgs[V, A](name, description, directives)
+
+  /**
+   * Creates a new hand-rolled schema. For normal usage use the derived schemas, this is primarily for schemas
+   * which can't be resolved by derivation.
+   * @param name The name of the type
+   * @param description An optional description of the type
+   * @param directives The directives to add to the type
+   * @param fields The fields to add to this object
+   *
+   * {{{
+   *  case class Group(id: String, users: UQuery[List[User]], parent: UQuery[Option[Group]], organization: UQuery[Organization])
+   *  case class Organization(id: String, groups: UQuery[List[Group]])
+   *  case class User(id: String, group: UQuery[Group])
+   *
+   *  implicit val groupSchema: Schema[Any, Group] = obj("Group", Some("A group of users"))(implicit ft =>
+   *    NonEmptyChunk(
+   *      field("id")(_.id),
+   *      field("users")(_.users),
+   *      field("parent")(_.parent),
+   *      field("organization")(_.organization)
+   *    )
+   *  )
+   *
+   *  implicit val orgSchema: Schema[Any, Organization] = obj("Organization", Some("An organization of groups"))(implicit ft =>
+   *    NonEmptyChunk(
+   *      field("id")(_.id),
+   *      field("groups")(_.groups)
+   *    )
+   *  )
+   *
+   *  implicit val userSchema: Schema[Any, User] = obj("User", Some("A user of the service"))(implicit ft =>
+   *    NonEmptyChunk(
+   *      field("id")(_.id),
+   *      field("group")(_.group)
+   *    )
+   *  )
+   *
+   * }}}
+   *
+   */
+  def obj[R1, V](name: String, description: Option[String] = None, directives: List[Directive] = Nil)(
+    fields: FieldType => NonEmptyChunk[(__Field, V => Step[R1])]
```

Comment:
How about using `List` instead of `NonEmptyChunk`? The schema validation will catch the case where an empty list is passed, and that type is friendlier to people who don't know ZIO. I don't have a strong opinion on this though, it's just that I know a bunch of people using Caliban without ZIO types.

Reply:
Fair point, I liked NEC because it enforces the behavior at compile time, but yes didn't account for people looking to use without `ZIO` types.